src/utilities/use-format-message.js

```diff
@@ -0,0 +1,27 @@
+import React, { Fragment } from 'react';
+const { useIntl } = require('react-intl');
+
+const useFormatMessage = () => {
+  const { formatMessage } = useIntl();
+  return (message, values = {}) => {
+    const internalValues = Object.entries(values).reduce(
+      (acc, [key, value]) => {
+        return {
+          ...acc,
+          [key]:
+            typeof value === 'function'
```

Comment:
What if you want to put an element there?

Reply:
Then its returned as it was. You can't inject key to cloned element since its not a prop anyway. I think we can make this a convention in our apps to just use function components.

Reply:
You can still wrap an element in the fragment too.

Reply:
Yes, you can but detecting an element is fairly ambiguous. The react element does not work very well if you remember when we have tried that in DDF. It all comes down to if you can call the value like `value()` or like `<value />`. We might just try the latter and see if it works though. :thinking: gimme a moment.

Reply:
I think checking `typeof x === 'object'` should be enough for this case. (As you can't put a standard object there.)

Reply:
True and even if it does fail, there is still the try-catch block which will catch the error anyway:shrug:. Here I am trying to think of stuff that won't happen...

Reply:
Nah it won't work. We are starting to get to some stupidly large overhead for what we try to achieve. The chunks parm is not your standard component props object.

```jsx
            typeof value === 'function' || typeof value === 'object'
              ? (chunks) => {
                  const Component = value;
                  return (
                    <Fragment key={key}>
                      <Component>
                        {chunks.map((chunk, index) => (
                          <Fragment key={index}>{chunk}</Fragment>
                        ))}
                      </Component>
                    </Fragment>
                  );
                }
              : value
```

this is just too much to make such a simple thing working. If we won't use class components (which we are not) we can do wich just the function type.